### PR TITLE
git[checkout]: show tags in checkout suggestions

### PR DESCRIPTION
**Pull request by fwesss**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature

**What is the current behavior? (You can also link to an open issue here)**
git checkout will list branches, files, and commits to checkout but not tags.

**What is the new behavior (if this is a feature change)?**
Tags will be listed in the suggestions for git checkout.

**Additional info:**
The current branch is normally given top priority with a star icon. However, if we are in a detached HEAD state after checking out a tag, I don't think we want to suggest the current branch. It would just show (detached HEAD state) and isn't actionable suggestion.

Re:https://github.com/withfig/autocomplete/issues/403

### CODE REVIEW

**Review by fedeci — COMMENTED** [review]
Maybe we should assign a higher priority to branches rather than commits or tags.

**Comment by Nutlope** [comment]
> Maybe we should assign a higher priority to branches rather than commits or tags.

I agree with @fedeci. Maybe branches > commits > tags?

**Comment by fwesss** [comment]
I've increased the priority given to branches in postProcessBranches.
Looks like the only time this generator is used cumulatively is when generating suggestions for tags so I don't think this change in prioritization would affect the results in any other commands.

I'm not a super git expert or 100% familiar with this spec though so any suggestions or comments are appreciated!

File: dev/git.ts [hunk 1]
```diff
@@ -3966,10 +3976,11 @@ export const completionSpec: Fig.Spec = {
       args: [
         {
           name: "branch or file",
```
Comment (Nutlope) [inline dev/git.ts:3978]: maybe change this to: "branch, file, or tag"?

**Comment by Nutlope** [comment]
> I've increased the priority given to branches in postProcessBranches.
> Looks like the only time this generator is used cumulatively is when generating suggestions for tags so I don't think this change in prioritization would affect the results in any other commands.

Just tested it out and it works great! Just added one more minor comment

File: dev/git.ts (hunk 1 above)
Comment (fwesss) [reply dev/git.ts]: Fixed!

**Comment by Nutlope** [comment]
LGTM!

**Pull request merged**